### ISSUE TITLE
Drei <Html transform /> flickering

### ISSUE THREAD

**Issue by henriandrade** [RodrigoHamuy/react-three-map#106]
React Drei <Html> component with property "transform={true}" seems to not work under react-three-map canvas.

**Comment by RodrigoHamuy** [comment]
Can you add a codesanbox to take a look? Thanks!
I suspect is related to the camera transform.

**Comment by tom-twinlabs** [comment]
I'm having a similar issue, only difference I see it without transform mode in drei/html . I made a sandbox [here](https://codesandbox.io/p/devbox/react-map-three-drei-html-xcwlw6?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyh6yfrv0007356no7h2873v%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyh6yfrv0002356nb29ncpt3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyh6yfrv0004356nncnuxom3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyh6yfrv0006356nuuxzzwks%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyh6yfrv0002356nb29ncpt3%2522%253A%257B%2522id%2522%253A%2522clyh6yfrv0002356nb29ncpt3%2522%252C%2522activeTabId%2522%253A%2522clyh7fb3f00f7356nzruw2153%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyh6yfrv0001356nr531ampg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%2522%252C%2522initialSelections%2522%253A%255B%255D%252C%2522id%2522%253A%2522clyh7fb3f00f7356nzruw2153%2522%252C%2522mode%2522%253A%2522temporary%2522%257D%255D%257D%252C%2522clyh6yfrv0006356nuuxzzwks%2522%253A%257B%2522id%2522%253A%2522clyh6yfrv0006356nuuxzzwks%2522%252C%2522activeTabId%2522%253A%2522clyh7cr6600b9356nqr9c5ghf%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clyh7cr6600b9356nqr9c5ghf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clyh6yfrv0004356nncnuxom3%2522%253A%257B%2522id%2522%253A%2522clyh6yfrv0004356nncnuxom3%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyh6zbqw003i356nazk9zf7h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clyh6zp9r00kddjip9pf4fh0k%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clyh6zbqw003i356nazk9zf7h%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) but I can't replicate it in the sandbox, I'll keep trying but feel free to collaborate on that sandbox

**Comment by RodrigoHamuy** [comment]
Yes, I guess if you use `transform` we may need to improve our camera matrix calculations or maybe we are not updating the viewports correctly.

**Comment by vespasianvs** [comment]
I know this is an older issue - not sure if it's related, but we've also been having problems with the Drei Html component. I've done a fair amount of tracking back of values to try to work out the problem. 

We're using the Canvas in overlay mode, which may or may not be a factor but basically in the syncCamera method, the calculation generates 'far' values for the camera that are invalid. The 'far' value should always be greater than the 'near' value, and it's fairly easy to get it into the massive negative number if you rotate the view. I've done a code sandbox [here ](https://codesandbox.io/p/devbox/react-map-three-drei-html-forked-l8spxf?workspaceId=ws_V5CSXkruYfEcnPMAwZxXoH) demonstrating the issue - zoom in and out is OK, but try rotating the camera. If you take it out of 'overlay' mode, you instead end up with a value of -Infinity, which again isn't a valid value. This then gives weird behaviour within the Drei HTML component whether in transform or without as it uses the camera.near and camera.far values to calculate z-index values (and the invalid camera.far value means that you get negative z-indexes.). For now we're using patch-package to just hardcode the recommended max for the camera.far to 2000
